zappa/zappa.py

```diff
@@ -697,13 +705,36 @@ def create_api_gateway_routes(self, lambda_arn, api_name=None, api_key_required=
         resource.PathPart = "{proxy+}"
         self.cf_template.add_resource(resource)
 
-        self.create_and_setup_methods(restapi, resource, lambda_arn, api_key_required,
-                                      integration_content_type_aliases, authorization_type, 1)  # pragma: no cover
+        self.create_and_setup_methods(restapi, resource, api_key_required, invocations_uri,
+                                      integration_content_type_aliases, authorization_type, authorizer_resource, 1)  # pragma: no cover
 
         return restapi
 
-    def create_and_setup_methods(self, restapi, resource, lambda_arn, api_key_required,
-                                 integration_content_type_aliases, authorization_type, depth):
+    def create_authorizer(self, restapi, uri, identity_source, authorizer_result_ttl, identity_validation_expression):
+        """
+        Create Authorizer for API gateway
+        """
+        if not self.credentials_arn:
+            self.get_credentials_arn()
+
+        authorizer_resource = troposphere.apigateway.Authorizer("Authorizer")
+        authorizer_resource.RestApiId = troposphere.Ref(restapi)
+        authorizer_resource.Name = "ZappaAuthorizer"
```

Comment:
Will this create a collision if you have multiple applications with authorizers on the same account?

Maybe should be project_name + "ZappaAuthorizer" ?